### PR TITLE
Update ApplicationEngine.py

### PR DESCRIPTION
**Pull request by jseagrave21**
**What current issue(s) does this address, or what feature is it adding?**
Add appropriate gas cost for `VERIFY`
Add appropriate GAS cost for `CHECKMULTISIG` for an array

To meet the requirement of issue #564 .

**How did you solve this problem?**
Interpreting

**How did you make sure your solution works?**
Individual testing

**Are there any special changes in the code that we should be aware of?**
No

**Please check the following, if applicable:**

- [ ] Did you add any tests?
- [x] Did you run `make lint`?
- [x] Did you run `make test`?
- [x] Are you making a PR to a feature branch or development rather than master?
- [x] Did you add an entry to `CHANGELOG.rst`? (if not, please do)

### CODE REVIEW

**Comment by jseagrave21** [comment]
Obviously this is not ready to merge. I will keep trying to figure out why Travis is failing. If I can't, I will close the PR.

One question I have is if my approach for determining if the item is an array using `isinstance` is valid. I am not sure what this data would look like.

File: neo/SmartContract/ApplicationEngine.py [hunk 1]
```diff
@@ -292,10 +292,19 @@ def GetPrice(self):
             return 20
         elif opcode == CHECKSIG:
```
Comment (localhuman) [inline neo/SmartContract/ApplicationEngine.py]: you could do `elif opcode in [CHECKSIG,VERIFY]:`

File: neo/SmartContract/ApplicationEngine.py [hunk 2]
```diff
@@ -292,10 +292,19 @@ def GetPrice(self):
...
+        elif opcode == VERIFY:
+            return 100
         elif opcode == CHECKMULTISIG:
             if self.EvaluationStack.Count == 0:
                 return 1
-            n = self.EvaluationStack.Peek().GetBigInteger()
+            
+            item = self.EvaluationStack.Peek()
+
+            array = isinstance(item, list)
```
Comment (localhuman) [inline neo/SmartContract/ApplicationEngine.py]: You'll need to check for the interop `Array` type, not the built in `list` type.
```
if isintance(item,Array):
   n = item.Count
```

Comment (jseagrave21) [reply neo/SmartContract/ApplicationEngine.py]: I can't believe it is as easy as just checking if it is an Array lol.

**Comment by jseagrave21** [comment]
@localhuman Thank you for the feedback! I was toying with the idea for checking for `Array` type, but I couldn't find any reference on Google, so I figured it couldn't be that easy. lol.

File: neo/SmartContract/ApplicationEngine.py [hunk 3]
```diff
@@ -290,12 +290,18 @@ def GetPrice(self):
...
             return 100
         elif opcode == CHECKMULTISIG:
             if self.EvaluationStack.Count == 0:
                 return 1
-            n = self.EvaluationStack.Peek().GetBigInteger()
+            
+            item = self.EvaluationStack.Peek()
+
+            if isinstance(item, Array):
+                n = item.count
```
Comment (localhuman) [inline neo/SmartContract/ApplicationEngine.py]: Should be `item.Count`

**Comment by localhuman** [comment]
Hmm, seems like there's something else going on here causing the tests to fail.  Will take a little deeper look...

**Comment by jseagrave21** [comment]
This was the same set of errors I was getting when I originally added `VERIFY`. I couldn't quickly identify why, which is why I originally commented about the Travis failures. I look also when I get back home, if you haven't figured it out by then.

**Comment by jseagrave21** [comment]
@localhuman Good find. I didn't think to import `VERIFY` since I didn't see it in the neo-project PR.

**Comment by jseagrave21** [comment]
@localhuman Thank you for working with me on this PR!

**Pull request merged**